### PR TITLE
Android GL Crash

### PR DESCRIPTION
**Pull request by bleege**
Tracking commits for updating the Android to Core GL integration for  #1435

### CODE REVIEW

File: src/mbgl/map/map_context.cpp [hunk 1]
```diff
@@ -230,8 +217,13 @@ void MapContext::render() {
     env.performCleanup();
 
     assert(style);
-    assert(painter);
 
+    if (!painter) {
+    painter = util::make_unique<Painter>(*spriteAtlas, *glyphAtlas, *lineAtlas);
```
Comment (tmpsantos) [inline src/mbgl/map/map_context.cpp]: Indentation is broken here, no?

Comment (ljbade) [inline src/mbgl/map/map_context.cpp]: Fixed, thanks.

**Comment by ljbade** [comment]
@jfirebaugh Can your review this and merge if happy once Travis passes (it should).

**Comment by ljbade** [comment]
Travis passed, ready for merging.

**Comment by bleege** [comment]
This looks mostly on the Android c++ side of the house so it should be good to merge in.

File: platform/ios/MGLMapView.mm [hunk 2]
```diff
@@ -272,7 +272,10 @@ - (BOOL)commonInit
     _mbglFileSource = new mbgl::DefaultFileSource([MGLFileCache obtainSharedCacheWithObject:self]);
 
     // Start paused on the IB canvas
-    _mbglMap = new mbgl::Map(*_mbglView, *_mbglFileSource, mbgl::MapMode::Continuous, _isTargetingInterfaceBuilder);
+    _mbglMap = new mbgl::Map(*_mbglView, *_mbglFileSource, mbgl::MapMode::Continuous);
+    if (_isTargetingInterfaceBuilder) {
+        _mbglMap->pause();
```
Comment (1ec5) [inline platform/ios/MGLMapView.mm]: Did you test this change in Interface Builder?

Comment (1ec5) [inline platform/ios/MGLMapView.mm]: OK, it does seem to work in Interface Builder designables, and `MGLMapView`s instantiated from storyboards seem to work fine.

**Pull request merged**